### DISCUSSION TITLE
Socket instance is not returning connected: true on "connection" and having a type error when it's called outside of provider

### DISCUSSION THREAD

**Discussion by erhanyasar** [socketio/socket.io#4921] [Q&A] [answered]
I'm using it through provider and I observe strange behaviour with it when I'm using it with Next.js 14.0.1 with TypeScript.

Even it does not fire connected event on "connection" with;
`socket.on("connection", (socket) => console.log("socket connected; ", socket));`

it does return and work with other methods followed like;
`socket.on("game:new", () => console.log("new game; "));`

The obvious thing i realized is the first logged event is `socket diconnected;  transport close` even the rest of the methods work successfully.

There's also another problem with my implementation. In socket provider, it's ok without any problem to make a call with;
` socket.on("game:set", () => console.log("set roulette; "));`

but when i share socket instance through provider and make the exact same call, I got the error;
`Argument of type '...' is not assignable to parameter of type 'ReservedOrUserEventNames<SocketReservedEvents, { (opts?: Partial<ManagerOptions & SocketOptions> | undefined): Socket<DefaultEventsMap, DefaultEventsMap>; (uri: string, opts?: Partial<...> | undefined): Socket<...>; }>'.ts(2345)`.

Any idea or insights would be appreciated.

PS: Currently socket context's type is `Socket<typeof DefaultEventsMap, typeof DefaultEventsMap>` while initializing it with `const SocketContext = createContext<SocketContextProps>(null!);`

**Comment by darrachequesne** [comment, marked as answer]
Hi! "connection" is on the server side, it's "connect" on the client side:

```js
socket.on("connect", () => console.log("socket connected; ", socket));
```

**Reply by erhanyasar** [reply]
Well, thank you!. Seems simple but tricky though
